src/Metadata/class-post-builder.php

```diff
@@ -271,25 +275,28 @@ private function build_metadata_post_times(): void {
 	 *
 	 * @since 3.3.0 Moved to class-metadata
 	 *
-	 * @param WP_Post              $post_obj The object for the post.
-	 * @param array<string, mixed> $parsely_options The parsely options.
+	 * @param WP_Post        $post_obj The object for the post.
+	 * @param ParselyOptions $parsely_options The parsely options.
+	 *
 	 * @return string Cleaned category name for the post in question.
 	 */
-	private function get_category_name( WP_Post $post_obj, array $parsely_options ): string {
+	private function get_category_name( WP_Post $post_obj, $parsely_options ): string {
 		$taxonomy_dropdown_choice = get_the_terms( $post_obj->ID, $parsely_options['custom_taxonomy_section'] );
 		// Get top-level taxonomy name for chosen taxonomy and assign to $parent_name; it will be used
 		// as the category value if 'use_top_level_cats' option is checked.
 		// Assign as the default category name if no value is checked for the chosen taxonomy.
-		$category_name = get_cat_name( get_option( 'default_category' ) );
-		if ( ! empty( $taxonomy_dropdown_choice ) && ! is_wp_error( $taxonomy_dropdown_choice ) ) {
+		$category_name = get_cat_name( get_default_category() );
+		if ( false !== $taxonomy_dropdown_choice && ! is_wp_error( $taxonomy_dropdown_choice ) ) {
 			if ( $parsely_options['use_top_level_cats'] ) {
 				$first_term = array_shift( $taxonomy_dropdown_choice );
-				$term_name  = $this->get_top_level_term( $first_term->term_id, $first_term->taxonomy );
+				if ( null !== $first_term ) {
+					$term_name = $this->get_top_level_term( $first_term->term_id, $first_term->taxonomy );
+				}
 			} else {
 				$term_name = $this->get_bottom_level_term( $post_obj->ID, $parsely_options['custom_taxonomy_section'] );
 			}
 
-			if ( is_string( $term_name ) && 0 < strlen( $term_name ) ) {
+			if ( isset( $term_name ) && is_string( $term_name ) && 0 < strlen( $term_name ) ) {
```

Comment:
Just a null check. Anywhere we are adding these checks are because PHPStan complains.

Will skip these type of comments below so please keep this reason in mind 🙂